db/system_keyspace.cc

```diff
@@ -1139,6 +1142,103 @@ schema_ptr system_keyspace::service_levels_v2() {
     return schema;
 }
 
+schema_ptr system_keyspace::roles() {
+    static thread_local auto schema = [] {
+        schema_builder builder(generate_legacy_id(NAME, ROLES), NAME, ROLES,
+        // partition key
+        {{"role", utf8_type}},
+        // clustering key
+        {},
+        // regular columns
+        {
+            {"can_login", boolean_type},
+            {"is_superuser", boolean_type},
+            {"member_of", set_type_impl::get_instance(utf8_type, true)},
+            {"salted_hash", utf8_type}
+        },
+        // static columns
+        {},
+        // regular column name type
+        utf8_type,
+        // comment
+        "roles for authentication and RBAC"
+        );
+        builder.with_version(system_keyspace::generate_schema_version(builder.uuid()));
```

Comment:
The new definitions are different: they are missing
```cpp
         builder.set_gc_grace_seconds(auth_gc_grace);
```

why?

Are there other differences that I missed? Why are the new definitions of these tables not copy-pastes of the old definitions?

The commit message only mentions that tables were moved from one keyspace to another, but the code seems to sneak-in some changes as well.

Reply:
that's the only change (in the code) but not in behavior because when it's part of the system it somehow gets correct gc grace seconds (default?)

Reply:
> somehow

I can't merge PRs when the author doesn't understand how things happen. "Somehow" is not a sufficient explanation.

Anyway I did the work, looked at the code:

```cpp
    schema_builder& set_gc_grace_seconds(int32_t gc_grace_seconds) {
        _raw._gc_grace_seconds = gc_grace_seconds;
        return *this;
    }
```

```cpp
    struct raw_schema {
...
        int32_t _gc_grace_seconds = DEFAULT_GC_GRACE_SECONDS;
```

```cpp
static constexpr int DEFAULT_GC_GRACE_SECONDS = 864000;
```

So it looks like the default **for any table** (not just the ones living in `system`) is 10 days

I found no code in `schema_builder` / `schema` that would give special treatment to `system` keyspace.

---

So judging from that, **there is a change in behavior** -- gc_grace was increased from 7 days to 10 days

Which is fine, but still, we should not introduce changes like these without being aware of them... This one turned out to be fine, but others could not